forc-pkg/src/pkg.rs

```diff
@@ -514,6 +514,35 @@ impl BuildPlan {
     pub fn compilation_order(&self) -> &[NodeIx] {
         &self.compilation_order
     }
+
+    /// Trim graph of the given build plan such that only given member and it's dependencies are
+    /// present in the graph. This will ensure that building the package only builds a part of the
+    /// workspace that belongs to the package itself.
+    ///
+    /// Returns the given build plan with the trimmed graph.
+    pub fn member_plan(&self, member_manifest: &PackageManifest) -> Result<Self> {
+        let mut graph = self.graph().clone();
+        let member_node = self
+            .member_nodes()
+            .find(|node_ix| graph[*node_ix].name == member_manifest.project.name)
+            .ok_or_else(|| {
+                anyhow!(
+                    "{} is not a member node in the dependency graph",
+                    member_manifest.project.name
+                )
+            })?;
+        let bfs = Bfs::new(&graph, member_node);
+        // Collect visitable nodes from the given node in the graph.
+        let visited_nodes: HashSet<NodeIx> = bfs.iter(&graph).collect();
+        graph.retain_nodes(|_, index| visited_nodes.contains(&index));
+        // Since some nodes are removed from the graph, compilation order needs to be reconstructed.
+        let compilation_order = compilation_order(&graph)?;
+        Ok(Self {
+            graph: graph.clone(),
+            manifest_map: self.manifest_map.clone(),
+            compilation_order,
+        })
+    }
```

Comment:
Niccceee, nailed it!

I think we can make this even more useful if we split it up into a couple of methods? E.g.

```rust
    /// Produce the node index of the member with the given name.
    pub fn find_member_index(&self, member_name: &str) -> Option<NodeIx> {
        // ...
    }

    /// Produce an iterator yielding indices for the given node and its dependencies in BFS order.
    pub fn node_deps(&self, n: NodeIx) -> impl '_ + Iterator<Item = NodeIx> {
        // ...
    }

    // Produce a new build plan with only the packages required to build the given node.
    pub fn node_plan(&self, n: NodeIx) -> Result<Self> {
        // ...
    }
```